
# HG changeset patch
# User Paul Adenot <paul@paul.cx>
# Date 1665476812 0
# Node ID 81715fef2264de0412ec868c22167a119103b81b
# Parent  75c1403f58f79d1abd43d33fdd1beb36db9367c6
Bug 1793289 - Override EXTERN_ASM when building ffvpx in generic / no asm configuration. r=alwu

Thanks to Ray Vine <ray-v@inbox.lv> for the bug report and a suggested fix.

Differential Revision: https://phabricator.services.mozilla.com/D158679

diff --git a/media/ffvpx/config_override.h b/media/ffvpx/config_override.h
--- a/media/ffvpx/config_override.h
+++ b/media/ffvpx/config_override.h
@@ -5,16 +5,23 @@
 
 #ifdef MOZ_LIBAV_FFT
 #undef CONFIG_FFT
 #undef CONFIG_RDFT
 #define CONFIG_FFT 1
 #define CONFIG_RDFT 1
 #endif
 
+// override '#define EXTERN_ASM _' in config_generic.h to allow building with
+// gcc on arm
+#if defined(__GNUC__) && defined(__arm__)
+#undef EXTERN_ASM
+#define EXTERN_ASM
+#endif
+
 #if defined(MOZ_WAYLAND) && !defined(MOZ_FFVPX_AUDIOONLY)
 #undef CONFIG_VAAPI
 #undef CONFIG_VAAPI_1
 #undef CONFIG_VP8_VAAPI_HWACCEL
 #undef CONFIG_VP9_VAAPI_HWACCEL
 #undef CONFIG_AV1_VAAPI_HWACCEL
 #define CONFIG_VAAPI 1
 #define CONFIG_VAAPI_1 1

